### PR TITLE
[RPW][Rendering Debugger]Cleanup of enum fields

### PR DESCRIPTION
**Pull request by alex-vazquez**
---
### Purpose of this PR
- Adding new API for Drawing DebugUI.Fields
- Making sure Enum fields work as expected
- Simplify code path for index change

---
### Testing status
- Undo/Redo
- Modify Enum fields values see that the value is properly set

### CODE REVIEW

**Review by JulienIgnace-Unity — APPROVED** [review]
Looks good to me but we need to make sure HDRP split enum debug items are still working properly :)

File: com.unity.render-pipelines.core/Runtime/Debugging/DebugUI.Fields.cs [hunk 1]
```diff
@@ -344,15 +349,24 @@ internal void InitQuickSeparators()
...
+            /// <summary>
+            /// Set the value of the field.
+            /// </summary>
+            /// <param name="value">Input value.</param>
+            public override void SetValue(int value)
             {
-                if (enumNames == null)
-                    enumNames = new GUIContent[0];
+                Assert.IsNotNull(setter);
+                var v = ValidateValue(value);
```
Comment (RSlysz) [inline com.unity.render-pipelines.core/Runtime/Debugging/DebugUI.Fields.cs:359]: try to not use abreviation if possible

**Review by RSlysz — COMMENTED** [review]
There is a lot of abbreviation:
- w
- s
- v
Normally it is not good but I see that there is a lot of them in this file. Can you try to remove them little by little as we do fix on this file? This is challenging readability.

File: com.unity.render-pipelines.core/Editor/Debugging/DebugUIDrawer.cs [hunk 2]
```diff
@@ -35,15 +36,25 @@ public class DebugUIDrawer
         protected T Cast<T>(object o)
             where T : class
         {
-            var casted = o as T;
+            if (o == null) return null;
```
[the remark this answers is not in the document]
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Editor/Debugging/DebugUIDrawer.cs:39]: The null exception was silent fialing here as there is a catch on the TraverseGUI method on the DebugWindow that is doing nothing with it, so the result is the same.

**Review by RSlysz — APPROVED** [review]

**Review by TomasKiniulis — APPROVED** [review]
Thanks! Works well now

**Comment by alex-vazquez** [comment]
Yamato is Green and QA validates the PR, merging...

**Pull request merged**